tooling/nargo_cli/src/cli/fmt_cmd.rs

```diff
@@ -22,24 +23,26 @@ pub(crate) struct FormatCommand {
     pub(super) package_options: PackageOptions,
 }
 
-pub(crate) fn run(args: FormatCommand, config: NargoConfig) -> Result<(), CliError> {
-    let check_mode = args.check;
+impl WorkspaceCommand for FormatCommand {
+    fn package_selection(&self) -> PackageSelection {
+        match self.package_options.package_selection() {
+            PackageSelection::DefaultOrAll => PackageSelection::All,
+            other => other,
+        }
+    }
 
-    let toml_path = get_package_manifest(&config.program_dir)?;
-    let selection = match args.package_options.package_selection() {
-        PackageSelection::DefaultOrAll => PackageSelection::All,
-        other => other,
-    };
-    let workspace = resolve_workspace_from_toml(
-        &toml_path,
-        selection,
-        Some(NOIR_ARTIFACT_VERSION_STRING.to_string()),
-    )?;
+    fn lock_type(&self) -> LockType {
+        // Writes source files, but doesn't touch compilation artifacts.
+        LockType::None
+    }
+}
 
+pub(crate) fn run(args: FormatCommand, workspace: Workspace) -> Result<(), CliError> {
+    let check_mode = args.check;
     let mut workspace_file_manager = workspace.new_file_manager();
     insert_all_files_for_workspace_into_file_manager(&workspace, &mut workspace_file_manager);
 
-    let config = nargo_fmt::Config::read(&config.program_dir)
+    let config = nargo_fmt::Config::read(&workspace.root_dir)
```

Comment:
I don't know if this is right. I think the closest `noirfmt.toml` file should be picked up so that you can customize formatting for a project inside a workspace differently from the entire workspace.

Reply:
Thanks for highlighting that, I thought about whether people want to do that too. The reason I thought this should be fine is because in my earlier PR I actually changed the `config.program_dir` based on whether `--workspace` or `--package` was present, which meant you had to `cd` into the package for the package specific configuration to be effective. If we want this to work reliably we need to parse the config in the loop, with a fallback to the workspace if it doesn't exist. 

I also thought that for anyone that uses a workspace, why would they actually not want all their codebase to be consistently formatted 🤷

Reply:
Sounds good.

Yesterday I found out that [formatting via LSP always uses the default configuration](https://github.com/noir-lang/noir/issues/7340) so I thought about fixing that first by passing the directory the file is in, then by recursively checking parent directories until a `noirfmt.toml` file is found. That's [how rustfmt works](https://github.com/rust-lang/rustfmt?tab=readme-ov-file#configuring-rustfmt). So when I pick up that issue I might modify this again (or, first I'll check what rustfmt does when you have a config file in the project and in the workspace).

Reply:
Cool! There's also the fact that unlike every other command, `fmt` changes `PackageSelection::DefaultOrAll` to `PackageSelection::All`, because we want to format the everything unless specifically told otherwise, but that also means what if we're in a subdirectory and don't select any `--package` or the `--workspace`, it will still be effectively `--workspace`. So yeah, I think what you said about looking for the closest `noirfmt.toml` visible in every package is the right way.